### PR TITLE
Pinning Julia versions 

### PR DESCRIPTION
**Pull request by gottacatchenall**
This currently this instantiates the project at the root, but each script is run in the  `/scripts/<SCRIPT_NAME` and uses that dir as the project by default. Easiest way around this would be to have `Pkg.activate("/")`, though there may be some long-term benefit in having individual `Project.toml`'s for each script that are all installed when the Julia docker image is built

### CODE REVIEW

**Comment by tpoisot** [comment]
> This currently this instantiates the project at the root, but each script is run in the /scripts/<SCRIPT_NAME and uses that dir as the project by default. 

We can also install the required packaged in the container's default (`@1.whatever`) environment and only add the required packages to each script's `Project.toml`. When starting julia with `--project`, the default environment remains accessible (I think?)

File: runners/julia-dockerfile [hunk 1]
```diff
@@ -1,6 +1,11 @@
 FROM julia:1.9.3
 
 # Pre-compiling Julia dependencies
-RUN  julia -e 'pwd(); using Pkg; Pkg.add.(["SpeciesDistributionToolkit", "Dates", "Clustering", "JSON", "CSV", "DataFrames", "StatsBase", "EvoTrees", "MultivariateStats" ]); Pkg.instantiate();'
+# RUN  julia -e 'pwd(); using Pkg; Pkg.add.(["SpeciesDistributionToolkit", "Dates", "Clustering", "JSON", "CSV", "DataFrames", "StatsBase", "EvoTrees", "MultivariateStats" ]); Pkg.instantiate();'
```
Comment (tpoisot) [inline runners/julia-dockerfile:4]: this should probably use https://pkgdocs.julialang.org/v1/api/#Pkg.PackageSpec to make sure we set the version of each package

**Review by jmlord — CHANGES_REQUESTED** [review]

**Comment by gottacatchenall** [comment]
I have added the scripts and made a version of instantiate that installs and precompiles and environment for each Julia script (right now only helloWorld.jl is on main)

**Comment by gottacatchenall** [comment]
i'll do this somewhere else :(

**Pull request closed without merging**